File: backend/app/adapters/ai_model/registry.py
from __future__ import annotations
"""AI 模型注册中心"""

from app.adapters.ai_model.base import AIModelAdapter


class ModelRegistry:
    _adapters: dict[str, AIModelAdapter] = {}
    _default: str | None = None

    @classmethod
    def register(cls, adapter: AIModelAdapter) -> None:
        cls._adapters[adapter.name] = adapter

    @classmethod
    def get(cls, name: str) -> AIModelAdapter | None:
        return cls._adapters.get(name)

    @classmethod
    def list(cls) -> list[AIModelAdapter]:
        return list(cls._adapters.values())

    @classmethod
    def set_default(cls, name: str) -> None:
        if name in cls._adapters:
            cls._default = name

    @classmethod
    def get_default(cls) -> AIModelAdapter | None:
        if cls._default:
            return cls._adapters.get(cls._default)
        return next(iter(cls._adapters.values()), None)


# Singleton
model_registry = ModelRegistry

File: frontend/src/lib/ai/registry.ts
import type { AIModelAdapter } from './model-adapter';

class ModelRegistry {
  private adapters: Map<string, AIModelAdapter> = new Map();
  private defaultModel: string | null = null;

  register(adapter: AIModelAdapter): void {
    this.adapters.set(adapter.name, adapter);
  }

  get(name: string): AIModelAdapter | undefined {
    return this.adapters.get(name);
  }

  list(): AIModelAdapter[] {
    return Array.from(this.adapters.values());
  }

  setDefault(name: string): void {
    if (this.adapters.has(name)) {
      this.defaultModel = name;
    }
  }

  getDefault(): AIModelAdapter | undefined {
    if (this.defaultModel) {
      return this.adapters.get(this.defaultModel);
    }
    return this.adapters.values().next().value;
  }
}

// Singleton instance
export const modelRegistry = new ModelRegistry();
